Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1549,12 +1553,30 @@ private static void MakeSharedOpaqueOutputIfNeeded(IPipExecutionEnvironment envi
                                 }
                             }
 
+                            if (userRetry)
+                            {
+                                counters.IncrementCounter(PipExecutorCounter.ProcessUserRetriesImpactedPips);
+                                if (result.Status == SandboxedProcessPipExecutionStatus.Succeeded)
+                                {
+                                    //TODO update PipsSucceedingAfterUserRetry
+                                }
+                                else if (result.Status == SandboxedProcessPipExecutionStatus.ExecutionFailed)
+                                {
+                                    //TODO update PipsFailingAfterLastUserRetry
+                                }
+                            }
+
                             return result;
                         }
                     });
 
             processExecutionResult.ReportSandboxedExecutionResult(executionResult);
 
+            if (executionResult.PipProperties != null && executionResult.PipProperties.Count > 0)
+            {
+                //TODO create and/or update global pip Properties counters

```

Comment:
>//TODO create and/or update global pip Properties counters [](start = 16, length = 58)

Do you want to collect/aggregate the pip properties for each retry or only the final one? In this current code, it seems that you're only interested in the final run.

Reply:
I want the count of pips with a given property.  As you said above, the result is already passed through to the scheduler (so I should be able to do the update there).  I do pass through the earlier dictionary of pip properties when re-running the pips, though I'll need to check if those are returning early prior to parsing out the existing properties.

Reply:
I've moved UpdatePipPropertiesCountsAsync earlier in the process, so that it counts the values before starting the retry as well.